crates/matrix-sdk-ui/src/timeline/tests/encryption.rs

```diff
@@ -153,17 +153,10 @@ async fn test_retry_message_decryption() {
     assert_eq!(message.body(), "It's a secret to everybody");
     assert!(!event.is_highlighted());
 
+    // The message should not be re-reported as a late decryption.
     {
         let utds = hook.utds.lock().unwrap();
-        assert_eq!(utds.len(), 2);
-
-        // The previous UTD report is still there.
-        assert_eq!(utds[0].event_id, event.event_id().unwrap());
-        assert!(utds[0].time_to_decrypt.is_none());
-
-        // The UTD is now *also* reported as a late-decryption event.
-        assert_eq!(utds[1].event_id, event.event_id().unwrap());
-        assert!(utds[1].time_to_decrypt.is_some());
```

Comment:
I removed it because testing that it's the same seems like we're just testing the test harness. (There is no mechanism by which an entry can ever be removed from `hook.utds`.) But sure, double-checking is cheap

Reply:
We don't know how the test is going to evolve, so having more checks is cautious for the future, and as you're saying it's cheap to keep.